preprocess.lua

```diff
@@ -275,6 +277,12 @@ local function main()
 
   onmt.utils.Opt.init(opt, requiredOptions)
 
+  local mute = (opt.log_file:len() > 0)
```

Comment:
This should ideally be factorized in a static `init` function inside the Logger which returns a new Logger object depending on the options.